internal/sink/file.go

```diff
@@ -0,0 +1,174 @@
+package sink
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/google/uuid"
+
+	"github.com/brexhq/substation/config"
+	"github.com/brexhq/substation/internal/errors"
+	"github.com/brexhq/substation/internal/log"
+)
+
+const (
+	// errFileEmptyPrefix is returned when the sink is configured with a prefix
+	// key, but the key is not found in the object or the key is empty.
+	errFileEmptyPrefix = errors.Error("empty prefix string")
```

Comment:
These error strings and their comments duplicate `errAWSS3EmptyPrefix` & `errAWSS3EmptySuffix`, could they be aggregated with `errPathEmptySuffix` or similar?